walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,117 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```

Comment:
This function has gotten pretty long in general. Once it's gotten to this size (it's debatable how big is too big, but a good rule of thumb is a 20-40 lines) you should really start considering break out more helper functions. 

You should have helper functions for generally two reasons:
(1) hold shared code to reduce duplication
(2) give meaningful names to functions so that they can be understood at a higher level

This would be an example of (2)

Reply:
Should I rename query()? That was the name provided to us in the tutorial.

Reply:
If there was production code, then I would probably encourage you to change the 'query' name, yeah. 

But since it's the function given for the purposes of the week's learning, it's fine to leave it (I assume it was left intentionally vague since you have to figure out the functionality).